lib/ollo.ex

```diff
@@ -1,18 +1,71 @@
 defmodule Ollo do
   @moduledoc """
-  Documentation for Ollo.
+  Ollo: Oauth2 Provider
   """
 
+  alias Ollo.Config
+
+  @doc """
+  Gets the client matching the client_id
+  Returns a client struct OR nil
+  """
+  def get_client(client_id) do
+    Config.persistence_module.get_client(client_id)
+  end
+
   @doc """
-  Hello world.
+  Registers a client
+  Returns {:ok, client_struct} or {:error, error_struct}
+  """
+  def register_client(%{name: name} = argv) do
+    Config.persistence_module.register_client(argv)
+  end
 
-  ## Examples
+  @doc """
+  Grants authorization to a user's account to a client
+  Returns {:ok, client_auth} or {:error, %{error: :error_atom}}
 
-      iex> Ollo.hello
-      :world
+  Verifies client_id is a valid client, and the scopes are in the allowed scopes
 
+  Does not verify the user_id
```

Comment:
should it verify the user_id here?

Reply:
I think this can be decided later.. perhaps it ought to, or perhaps the Persistence logic that the user of the package specifies should handle this